src/bridge/IrcBridge.ts

```diff
@@ -348,6 +354,15 @@ export class IrcBridge {
                 ircHandlerCalls.inc({method: kv[0]}, kv[1]);
             });
         });
+
+        metrics.addCollector(async () => {
+            clientStates.reset();
+            for (const [server, states] of Object.entries(this.clientPool.getConnectionStatesForAllServers())) {
+                for (const [state, value] of Object.entries(states)) {
```

Comment:
this is a lot of looping, four levels deep at this point to get the data in the right shape. anything that can be done here?

Reply:
Decided to save two loops and expose the gauge in the called function. Better?